### PR TITLE
Fix the IOMMU enabling issue in the default EFI boot mode

### PR DESCRIPTION
**Pull request by lucassong-mh**
Background: https://github.com/asterinas/asterinas/issues/1517

### CODE REVIEW

File: tools/qemu_args.sh [hunk 1]
```diff
@@ -50,8 +50,9 @@ if [ "$1" = "iommu" ]; then
     IOMMU_DEV_EXTRA=",iommu_platform=on,ats=on"
     IOMMU_EXTRA_ARGS="\
         -device intel-iommu,intremap=on,device-iotlb=on \
-        -device ioh3420,id=pcie.0,chassis=1\
+        -device ioh3420,id=pcie.0,chassis=1 \
```
Comment (tatetian) [inline tools/qemu_args.sh:53]: So the IOMMU enabling problem is caused by the lack of whitespace at the end?

Comment (lucassong-mh) [reply tools/qemu_args.sh:53]: NoNo, it's the lack of OVMF, elaberated in https://github.com/asterinas/asterinas/issues/1517:

> I was wondering if we must use BIOS to enable IOMMU, why EFI can't?? After some search and chatgpt, it turns out we must use OVMF firmware to boot, and currently we don't configure OVMF upon enabling IOMMU......
>
> So the approach is simple, whether we add `-bios /usr/share/OVMF/OVMF_CODE.fd` in `tools/qemu_args.sh`, or just modify the part in `OSDK.toml` to `qemu.args = "$(./tools/qemu_args.sh iommu -ovmf)"`, no need to worry about BIOS any more...

Comment (tatetian) [reply tools/qemu_args.sh:53]: Ok. It took you quite some time to find out that `-ovmf` is needed to enable IOMMU. Could you add a comment here to share you newly-gained knowledge?

```toml
# Explain why add -ovmf is important
qemu.args = "$(./tools/qemu_args.sh iommu -ovmf)"
```

File: tools/qemu_args.sh [hunk 2]
```diff
@@ -50,8 +50,9 @@ if [ "$1" = "iommu" ]; then
...
     "
+    # TODO: Add support for enabling IOMMU on AMD platforms
```
Comment (junyang-zh) [inline tools/qemu_args.sh:55]: I am wondering if you are suggesting that there's issues with this script running on AMD machines; or if our kernel don't support AMD-IOMMU?

Comment (lucassong-mh) [reply tools/qemu_args.sh:55]: Yes, we need to further verify whether our kernel functions correctly with IOMMU enabled on AMD machines. Basically the qemu command `-device intel-iommu` in the script should be changed to `-device amd-iommu`.

File: OSDK.toml [hunk 3]
```diff
@@ -40,7 +40,8 @@ qemu.args = "$(./tools/qemu_args.sh microvm)"
 
 [scheme."iommu"]
 supported_archs = ["x86_64"]
-qemu.args = "$(./tools/qemu_args.sh iommu)"
+# OVMF provides the firmware necessary for supporting IOMMU in the default EFI boot mode.
+qemu.args = "$(./tools/qemu_args.sh iommu -ovmf)"
```
Comment (tatetian) [inline OSDK.toml:44]: According to this comment, IOMMU requires OVMF. Then, why not let `qemu_args` enable OVMF automatically?

Comment (lucassong-mh) [reply OSDK.toml:44]: My first intention is both `iommu` and `-ovmf` are readily and seperately script parameters that can be used directly and conveniently.
Integrating OVMF arguments with IOMMU directly without exposing `-ovmf` is also feasible, and maybe better. I will modify it.

Comment (lucassong-mh) [reply OSDK.toml:43]: After I looked carefully at `OSDK.toml` and `qemu_args.sh`, I may have different opinions. We only utilize `qemu_args.sh` in four cases in `OSDK.toml`:

```
...

[qemu]
args = "$(./tools/qemu_args.sh normal -ovmf)"

...

[test.boot]
method = "qemu-direct"

[test.qemu]
args = "$(./tools/qemu_args.sh test)"

...

[scheme."microvm"]
boot.method = "qemu-direct"
build.strip_elf = true
qemu.args = "$(./tools/qemu_args.sh microvm)"

...

[scheme."iommu"]
supported_archs = ["x86_64"]
qemu.args = "$(./tools/qemu_args.sh iommu -ovmf)"

...
```

It is showed that we must enable OVMF in the default EFI boot mode ("normal" and "iommu"). And in the "qemu-direct" boot mode, OVMF is no longer needed and disabled.

So in this PR, we either just write `./tools/qemu_args.sh iommu -ovmf` without any comments. It indicates this scheme is similar to the default `./tools/qemu_args.sh normal -ovmf` that we use the EFI boot mode. Or we need to hide OVMF-related commands into `qemu_args.sh` without exposing `-ovmf`.

Comment (lucassong-mh) [reply OSDK.toml:43]: The changes in https://github.com/asterinas/asterinas/pull/1553 appear to address the issue I mentioned earlier. This PR can be based on that.

Comment (junyang-zh) [reply OSDK.toml:43]: Yep, #1553 , though intended to solve a different problem, can provide a better solution for this.

Comment (tatetian) [reply OSDK.toml:43]: So let's merge #1553 first, then rebase this PR on it?

Comment (lucassong-mh) [reply OSDK.toml:43]: Yes, I think so.

Comment (tatetian) [reply OSDK.toml:43]: #1533 merged

Comment (lucassong-mh) [reply OSDK.toml]: Revised and pushed

**Comment by lucassong-mh** [comment]
@boterinas rerun

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**